x-pack/plugins/maps/public/embeddable/has_map_config.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  HasParentApi,
+  HasType,
+  PublishesDataViews,
+  PublishesPanelTitle,
+  PublishesLocalUnifiedSearch,
+} from '@kbn/presentation-publishing';
+import { apiIsOfType } from '@kbn/presentation-publishing';
+import type { ILayer } from '../classes/layers/layer';
+
+export type HasMapConfig = HasType<'map'> & {
+  getLayerList: () => ILayer[];
+};
+
+export type MapApi = HasMapConfig &
+  PublishesDataViews &
+  PublishesPanelTitle &
+  PublishesLocalUnifiedSearch &
+  Partial<HasParentApi<unknown>>;
+
+export const apiHasMapConfig = (api: unknown): api is MapApi => {
```

Comment:
same issue as above. 

We need to be very careful to keep what our type guards check in sync with the type they narrow to.

Reply:
resolved with https://github.com/elastic/kibana/pull/176869/commits/c8c6a4c4e1c8ca2a783cff5ad77d015644eaa09e